app/models/booking.rb

```diff
@@ -7,7 +7,12 @@ class Booking < ApplicationRecord
   delegate :code, :type_yard, to: :yard, prefix: true
 
   enum status: {pending: 0, accept: 1, rejected: 2, cancel: 3}
+
   scope :status_asc, ->{order status: :asc}
+  scope :find_date_booking, (lambda do |param, date|
+    where("time_cost_id IN (?) AND booking_date = ?
+    AND status = 1", param, date).pluck :time_cost_id
```

Comment:
` status = 1` -> giá trị 1 em cũng nên lấy từ enum status em nhé

`.pluck :time_cost_id` xóa đi, gọi ở controller đi em, để scope này còn tái sử dụng dc

Reply:
Chi ơi, cái giá trị status em gắn giá trị nó bằng enum trong câu query nó không nhận chị ơi, theo em nghĩ là câu truy vấn này nó truy vấn vào database của sql, mà trong sql giá trị của nó là int, nên em nghĩ nó không hiểu giá trị từ enum, còn giá trị enum mà em truy vấn theo kiểu association thì nó vẫn hiểu

Reply:
- param là gì em nhỉ? đổi tên dễ đọc hơn xí được ko?
- `AND status = 1` em bỏ đoạn này cũng được nhé. Em chỉ cần query kiểu như này 
```rb
Booking.accept.find_date_booking(@yards.pluck(:id), params[:date]).pluck(:time_cost_id)
```
=> `pending: 0, accept: 1, rejected: 2, cancel: 3` là giá trị enum nên  em có thể gọi các giá trị enum như một scope